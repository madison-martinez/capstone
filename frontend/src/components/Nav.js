// import { USERS_API_URL } from '../utils/index';
import Link from 'next/link';
import { useEffect } from 'react';
import NavStyles from './styles/NavStyles';
import { useUser } from './User';

function Nav() {
  const currentUser = useUser();

  useEffect(() => {

  }, []);

  return (
    <NavStyles>
      {currentUser && (
        <>
          <Link href="/marketplace">
            <a>Marketplace</a>
          </Link>
          <Link href="/sell">
            <a>Sell</a>
          </Link>
          <Link href="/orders">
            <a>Orders</a>
          </Link>
          <Link href="/account">
            <a>Account</a>
          </Link>
          <Link href="/login">
            <a>Logout</a>
          </Link>
        </>
      )
      }
      {!currentUser && (
        <>
          <Link href="/signup">
            <a>Signup</a>
          </Link>
          <Link href="/login">
            <a>Login</a>
          </Link>

        </>
      )}
    </NavStyles>
  )
};

export default Nav;

