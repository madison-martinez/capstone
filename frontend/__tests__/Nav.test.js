import { render } from '@testing-library/react';
import Nav from '../src/components/Nav';

describe('<Nav />', () => {
    it('renders nav links', () => {
        const { container } = render(
            <Nav />
        );
        expect(container).toHaveTextContent('Marketplace');
        expect(container).toHaveTextContent('Sell');
    })
});