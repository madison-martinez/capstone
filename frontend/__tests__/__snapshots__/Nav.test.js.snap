// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders nav links if user is not logged in 1`] = `
<div>
  <ul
    class="NavStyles-sc-1429cmc-0 fLfJtV"
  />
</div>
`;
